general/v0.1/protocol_basics.md

```diff
@@ -37,40 +36,54 @@ Communication between fediverse server and FASP MUST use HTTPS in production
 or production-like settings. This requirement MAY be relaxed in
 development environments.
 
-Registration tokens are JSON Web Tokens (JWT) as defined in
-[RFC-7519](https://datatracker.ietf.org/doc/html/rfc7519).
-
-For authentication and authorization of API calls, FASP and fediverse server
-use the OAuth 2.0 protocol as defined in
-[RFC-6749](https://tool.ietf.org/html/rfc6749.html).
-
 Custom API calls are HTTPS calls sending, if necessary, JSON data
 (`Content-Type: application/json`) and receiving JSON data.
 
-### OAuth2, Authentication and Authorization
+### Request Integrity
+
+In order to allow both parties to verify the integrity of message
+contents, all requests MUST contain a `content-digest` HTTP header as
+defined by [RFC-9530](https://tools.ietf.org/html/rfc9530.html).
+
+The hashing algorith used MUST be SHA-256.
+
+### Authentication
 
 As described in [03: Registration](registration.md) both FASP and
-fediverse server use OAuth 2.0 to authorize API calls. Both MUST obtain a valid
-access token and send this as a "bearer token" in the `Authorization`
-HTTP header with every API call.
+fediverse server exchange client IDs and public keys. API requests are
+being authenticated by HTTP Message Signatures as defined in
+[RFC-9421](https://tools.ietf.org/html/rfc9421.html).
+
+The signature algorithm used is EdDSA Using Curve edwards25519.
+Signatures cover signature parameters, the derived components `@method`
+and `@target-uri` and the HTTP header `content-digest` (see previous
+section).
 
-Example header:
+The `keyid` parameter MUST include the client ID and the corrsponsing
+private key is used to generate the signature.
+
+The required signature parameters are `created` and `keyid`.
+
+Example headers:
 
 ```http
-Authorization: Bearer SpBr6rheOp891mwWOfT6Pb"
+Signature-Input: sig1=("@method" "@target-uri" "content-digest"); created=1728467285;
```

Comment:
Yes, probably. I thought about it, but left it out to be able to move more quickly. The actual signature in the next line is also a placeholder and not a real one (I mean, this is not a full example, so it could never be an actual real one, but in the initial draft I made an approximation of a real one, this one is just random gibberish).

I will improve this example before we merge this (if we decide to do that).